### DISCUSSION TITLE
1:1 calling in a group chat shared by multiple people

### DISCUSSION THREAD

**Discussion by Half-Shot** [element-hq/element-meta#684] [Ideas]
### Description

The vast majority of my important long standing conversations have both my HS account and my backup matrix.org account in them to ensure I can reach people in case of a failure. However, this stops me from being able to 1:1 call them in that room. There needs to be a good solution so that I don't need to make two rooms to do this.

### Steps to reproduce

- Open/Use a room with multiple userids, but two physical *people*.
- Make that room direct.
- Try to make call
- Call is automatically made into a group call.

**Comment by uhoreg** [comment]
Somewhat related to https://github.com/vector-im/riot-web/issues/3087#issuecomment-282410205 and https://github.com/vector-im/riot-web/issues/5571

**Comment by Half-Shot** [comment]
Mulled on this some more. If we label a room as a direct room, we could also allow the client to select a call target. This wouldn't confuse things too much and cause issues in a group room, it probably would be the expected behaviour

**Comment by t3chguy** [comment]
but in the end we'd still be firing an `m.call` into the room that technically anyone could pick up?

**Comment by Half-Shot** [comment]
>but in the end we'd still be firing an m.call into the room that technically anyone could pick up?

Arguably by having a room with the direct flag we are indicating that we expect it to only involve two people, no?

Even if this wasn't the case, the person who picks up will send a pickup event into the room so we can tell who picked up. I can't see too much risk to this strategy?

**Comment by lampholder** [comment]
I find the current 1:1 vs conference UX pretty confusing, and I'd be afraid that changing the behaviour of the call button based on the direct flag could perhaps make things worse :\

I certainly agree though that the 1:1 calling behaviour is useful enough that it seems a shame to restrict it to very specific room configurations.

**Comment by Half-Shot** [comment]
@lampholder I think we are in agreement that the problem exists (and no matter how good the conference service, it never quite matches up to 1:1 calls), but I'm just throwing ideas at a wall to determine how we can do it.

**Comment by lampholder** [comment]
More thinkings:

I wonder if we could do better with some fairly simple changes. Perhaps we could:
- give the call button a different appearance (and tooltip) based on whether it's going to initiate a 1:1 call vs a conference call
- have the default behaviour remain that in a group chat it will initiate a conference and in a room with two non-bot participants it will use 1:1
- allow users to pick a 1:1 call target by clicking in the member list panel

The final bullet is hugely problematic. If the member profile panel includes an 'initate 1:1 call' button, I would expect that _not_ to leak any information about that call to the room I'm in. However, we rely on their being _a_ room to handle the call setup signalling. We _can_ mitigate this with elaborate UX, but it strikes me as an unfortunately complex and clumsy concept to express - if Riot told me when I went to intiate a 1:1 call in a group room that the call setup would be recorded in the room and visible to all room participants, my first thought would be 'why?'

This could be sidestepped if we used a dedicated room to handle call signalling between two users. This wouldn't need to appear as a chat room in the riot UX, though I suspect many people might find that approach distasteful :\

It also seems to me that having a user-visible record of the call details is important (time, participants, duration, etc.), but a UX question would be: how important is it that the call details reside in the room from which the 1:1 call was initated vs being exposed via a separate part of the UI?

**Comment by kittykat** [comment]
Moving this issue to discussions in Element meta as we need to make a cross platform decision on how to proceed :+1:
